### PR TITLE
Fixed dependency checker in detecting missing, unused or misplaced ones from JS, CSS and `package.json` files

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix (tests): Improved the dependency checker in detecting missing, unused, or misplaced packages from JS, CSS, and `package.json` files. Closes ckeditor/ckeditor5#8817.

---

### Additional information

Related PR is https://github.com/ckeditor/ckeditor5/pull/8854, which fixes errors found by this improved dependency checker.

The dependency checker is able to detect the new DLL-style imports, like `import { Plugin } from 'ckeditor5/src/core';`.
The `depcheck` version has been bumped to latest v1.3.1.

Fixes:
* Unused dependencies are now properly detected in JS or `package.json` files.
* Dependencies from CSS files are now properly detected and counted in the results.

Improvements:
* Misplaced dependencies `dependencies`  ⮀  `devDependencies` are now detected and reported.

### CODE REVIEW

**Comment by pomek** [comment]
I wonder whether we could use your changes on the current `master` branch in CKEditor 5. Could you rebase your branch with the `master` branch and change the base in the PR?

File: packages/ckeditor5-dev-tests/bin/check-dependencies.js [hunk 1]
```diff
@@ -13,134 +13,164 @@ const glob = require( 'glob' );
...
-		'*.vue': depCheck.parser.vue
-	},
-	ignoreDirs: [ 'docs', 'build' ],
-	ignoreMatches: [ 'eslint', 'eslint-plugin-ckeditor5-rules', 'eslint-plugin-mocha', 'husky', 'lint-staged', 'webpack-cli' ]
-};
-
-if ( Array.isArray( packageJson.depcheckIgnore ) ) {
-	depCheckOptions.ignoreMatches.push( ...packageJson.depcheckIgnore );
-}
+const packagePaths = getPackagePaths( process.argv.slice( 2 ) );
```
Comment (pomek) [inline packages/ckeditor5-dev-tests/bin/check-dependencies.js:16]: As I can see, there is a breaking change. In the previous version, if you missed the package directory, the CWD was used instead.

File: packages/ckeditor5-dev-tests/bin/check-dependencies.js [hunk 2]
```diff
@@ -13,134 +13,164 @@ const glob = require( 'glob' );
...
+}
 
-		if ( data[ 1 ] ) {
-			console.log( chalk.red( 'Missing dependencies:' ) );
-			console.log( data[ 1 ] + '\n' );
-		}
+/**
+ * Checks dependencies in provided package. If the folder does not contain a package.json file the function quits with success.
+ *
+ * @param {String} packagePath Relative path to package.
```
Comment (pomek) [inline packages/ckeditor5-dev-tests/bin/check-dependencies.js:39]: Missing the `@returns` annotation.

File: packages/ckeditor5-dev-tests/bin/check-dependencies.js [hunk 3]
```diff
@@ -13,134 +13,164 @@ const glob = require( 'glob' );
...
+ */
+async function checkDependenciesInPackage( packagePath ) {
+	const packageAbsolutePath = path.resolve( packagePath );
+	const packageJsonPath = path.join( packageAbsolutePath, 'package.json' );
 
-		if ( data[ 2 ] ) {
-			console.log( chalk.red( 'Missing devDependencies:' ) );
-			console.log( data[ 2 ] + '\n' );
-		}
+	if ( !fs.existsSync( packageJsonPath ) ) {
```
Comment (pomek) [inline packages/ckeditor5-dev-tests/bin/check-dependencies.js:45]: I'd display a warning message here. There must be something wrong with the current setup if the `package.json` file is missing.

**Review by pomek — CHANGES_REQUESTED** [review]
Minor improvements.

File: packages/ckeditor5-dev-tests/bin/check-dependencies.js (hunk 1 above)
Comment (pomek) [reply packages/ckeditor5-dev-tests/bin/check-dependencies.js:16]: But our CI will work with this version so we can leave it.

File: packages/ckeditor5-dev-tests/bin/check-dependencies.js (hunk 2 above)
Comment (psmyrek) [reply packages/ckeditor5-dev-tests/bin/check-dependencies.js:39]: Done.

File: packages/ckeditor5-dev-tests/bin/check-dependencies.js (hunk 3 above)
Comment (psmyrek) [reply packages/ckeditor5-dev-tests/bin/check-dependencies.js:46]: Warning has been added.

The CI calls `ls packages -1` to list all packages. The reason why I added this condition here is that after building the DLL, in the `packages/` path 4 new directories are created:

- `packages/ckeditor5-dll-balloon`
- `packages/ckeditor5-dll-classic`
- `packages/ckeditor5-dll-decoupled-document`
- `packages/ckeditor5-dll-inline`

which do not contain `package.json` files (they are just the output from the DLL build procedure and they contain only the built editor).

Comment (pomek) [reply packages/ckeditor5-dev-tests/bin/check-dependencies.js:46]: I guess we removed those packages a few days ago. Check whether you can delete those from your env.
Anyway, the warning will help with finding invalid packages so it's good to have it.

**Pull request merged**